### ISSUE TITLE
masking with cnn 

### ISSUE THREAD

**Issue by sungyongs** [fchollet/keras#3690]
Hi, I am working on NLP problems now.
I have character-level indices for certain texts.
For example, [1,16,16,12,5] = 'apple' (a->1, p->16, l->12, e->5)

Because of varying length of texts, I put zero padding in front of the indices.
For example, if the text length is fixed as 5, 
[1,16,16,12,5] = 'apple'
[0,0,2,1,4] = 'bad'

In this setting, I have several questions.
1) Would you help me to connect CNN after Embedding layer?
So, the input(e.g., [0,0,2,1,4]) is going to connect with Embedding layer first.
Then, the output of Embedding layer is going to connect with CNN layer.
However, CNN doesn't support "mask" so I couldn't on "mask_zero=True" at the Embedding layer.

2) I want to ignore output elements from zero padded input.
For example, if I use 'filter_length=2' CNN, the output is following:
[0,0,2,1,4]->Embedding->"real-valued vectors"->CNN->cnn_output
cnn_output.shape = (1, 4, nb_filters)
4 = length(5) - filter_length(2) + 1
cnn_output[0] <- CNN(Embedding([0,0]))
cnn_output[1] <- CNN(Embedding([0,2]))
cnn_output[2] <- CNN(Embedding([2,1]))
cnn_output[3] <- CNN(Embedding([1,4]))

However, as you see, the first vector of 4 output vectors is from zero-padding index.
So, I want to ignore cnn_output[0] when it is connected with following LSTM.

Would you have idea how to implement this idea?

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
